src/core/OriginalsSDK.ts

```diff
@@ -12,6 +12,17 @@ export class OriginalsSDK {
   public readonly bitcoin: BitcoinManager;
 
   constructor(config: OriginalsConfig) {
+    // Input validation
+    if (!config || typeof config !== 'object') {
+      throw new Error('Configuration object is required');
+    }
+    if (!config.network || !['mainnet', 'testnet', 'regtest'].includes(config.network)) {
```

Comment:
@cursoragent we also should be able to use `signet`. We probably need to make this change elsewhere in the codebase but we need signet everywhere